src/snapred/backend/service/ReductionService.py

```diff
@@ -341,30 +380,61 @@ def fetchReductionGroceries(self, request: ReductionRequest) -> Dict[str, Any]:
 
         # As an interim solution: set the request "versions" field to the latest calibration and normalization versions.
         #   TODO: set these when the request is initially generated.
-        calVersion = None
-        normVersion = None
-        calVersion = self.dataFactoryService.getThisOrLatestCalibrationVersion(request.runNumber, request.useLiteMode)
-        self.groceryClerk.name("diffcalWorkspace").diffcal_table(request.runNumber, calVersion).useLiteMode(
-            request.useLiteMode
-        ).add()
-
-        if ContinueWarning.Type.MISSING_NORMALIZATION not in request.continueFlags:
-            normVersion = self.dataFactoryService.getThisOrLatestNormalizationVersion(
+        if request.artificialNormalization is not None:
+            calVersion = None
+            normVersion = 0
+            calVersion = self.dataFactoryService.getThisOrLatestCalibrationVersion(
                 request.runNumber, request.useLiteMode
             )
-            self.groceryClerk.name("normalizationWorkspace").normalization(request.runNumber, normVersion).useLiteMode(
+            self.groceryClerk.name("diffcalWorkspace").diffcal_table(request.runNumber, calVersion).useLiteMode(
                 request.useLiteMode
             ).add()
+            return self.groceryService.fetchGroceryDict(
+                groceryDict=self.groceryClerk.buildDict(),
+                normalizationWorkspace=request.artificialNormalization,
+                **({"maskWorkspace": combinedMask} if combinedMask else {}),
+            )
 
-        request.versions = Versions(
-            calVersion,
-            normVersion,
-        )
+        else:
+            calVersion = None
+            normVersion = None
+            calVersion = self.dataFactoryService.getThisOrLatestCalibrationVersion(
+                request.runNumber, request.useLiteMode
+            )
+            self.groceryClerk.name("diffcalWorkspace").diffcal_table(request.runNumber, calVersion).useLiteMode(
+                request.useLiteMode
+            ).add()
```

Comment:
this is very similar code to [the above](https://github.com/neutrons/SNAPRed/pull/463/files#diff-8f698031b2852fad83d059775e7f13b373b64302000cb3f5dc92d1556c0f6ed0R384-R391),  I dont think `normVersion` is used in either of the highlighted snippets, so we can extract this duplicate code out and above the if/else statement.